src/main/java/org/testng/internal/AbstractParallelWorker.java

```diff
@@ -0,0 +1,116 @@
+package org.testng.internal;
+
+import org.testng.ClassMethodMap;
+import org.testng.IClassListener;
+import org.testng.ITestContext;
+import org.testng.ITestNGMethod;
+import org.testng.collections.Lists;
+import org.testng.internal.annotations.IAnnotationFinder;
+import org.testng.internal.thread.graph.IWorker;
+import org.testng.xml.XmlSuite;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public abstract class AbstractParallelWorker {
+
+    public static AbstractParallelWorker newWorker(XmlSuite.ParallelMode mode) {
+        if (XmlSuite.ParallelMode.INSTANCES.equals(mode)) {
+            return new InstanceBasedParallelParallelWorker();
+        }
+        return new ClassBasedParallelParallelWorker();
```

Comment:
Two "parallel"?

Reply:
Fixed. Silly me. Dont know how that one slipped through.